presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyNoUnresolvedSymbolExpression.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.sanity;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.expressions.DefaultRowExpressionTraversalVisitor;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.spi.WarningCollector;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.UnresolvedSymbolExpression;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.ExpressionExtractor;
+import com.facebook.presto.sql.planner.TypeProvider;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.lang.String.join;
+
+public final class VerifyNoUnresolvedSymbolExpression
+        implements PlanChecker.Checker
+{
+    @Override
+    public void validate(PlanNode plan, Session session, Metadata metadata, SqlParser sqlParser, TypeProvider types, WarningCollector warningCollector)
+    {
+        List<RowExpression> expressions = ExpressionExtractor.extractExpressions(plan)
+                .stream()
+                .collect(toImmutableList());
+        for (RowExpression expression : expressions) {
+            expression.accept(
+                    new DefaultRowExpressionTraversalVisitor<Void>()
+                    {
+                        @Override
+                        public Void visitUnresolvedSymbolExpression(UnresolvedSymbolExpression node, Void context)
+                        {
+                            throw new IllegalStateException(format("Unexpected UnresolvedSymbolExpression in logical plan: %s", join(".", node.getName())));
```

Comment:
Do we need join call here? I'm seeing toString method not using it:

```
    @Override
    public String toString()
    {
        return format("UnresolvedSymbolExpression{%s, %s}", type, name);
    }
```

Reply:
Yes, "a.b" is more user friendly than ["a", "b"] here. I will change toString() method later.